### PR TITLE
Don't use setuptools_scm from Sphinx itself

### LINKED ISSUE #1846: The latest and the stable documentation may have been built using the development version
Shouldn't be [there](https://pip-tools.readthedocs.io/) the latest released version?

Screenshot 2023-04-07 at 01 17 17

### PR DESCRIPTION
**Pull request by atugushev**
Perhaps resolves #1846.

> It is discouraged to use `setuptools_scm` from Sphinx itself, instead use `importlib.metadata` after editable/real installation. The underlying reason is, that services like Read the Docs sometimes change the working directory for good reasons and using the installed metadata prevents using needless volatile data there.

https://github.com/pypa/setuptools_scm/#usage-from-sphinx

##### Contributor checklist

- [ ] Provided the tests for the changes.
- [x] Assure PR title is short, clear, and good to be included in the user-oriented changelog

##### Maintainer checklist

- [x] Assure one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -20,18 +17,14 @@
...
-
 # The full version, including alpha/beta/rc tags
-release = get_scm_version()
+release = get_version(project)
+
+# The short X.Y version
+version = ".".join(release.split(".")[:3])
 
+print("pip-tools version:", version)
+print("pip-tools release:", release)
```
Comment (chrysle) [inline docs/conf.py:27]: Accidentally left over?

Comment (atugushev) [reply docs/conf.py:27]: Left on purpose to see versions in logs:
```
$ python -m sphinx ...
Running Sphinx v6.2.1
pip-tools version: 7.0.1
pip-tools release: 7.0.1.dev4
making output directory... done
...
```

Comment (chrysle) [reply docs/conf.py:27]: Maybe a comment about that could be useful?

**Review by chrysle — APPROVED** [review]
Looks good to me, then! Apologies I didn't get to this sooner.

**Review by RonnyPfannschmidt — APPROVED** [review]
lovely, this makes things more robust

i still would recommend to use the default local scheme but i can understand that the current behaviour is a pain for test-pypi

the next major version of setuptools_scm will provide a better behaviour on ci and the primary branches

File: docs/conf.py (hunk 1 above)
Comment (atugushev) [reply docs/conf.py]: Addressed in a49c505c435d3e7e42bd2f0d8ed7ebdbd2701d42.

**Comment by atugushev** [comment]
Thanks @RonnyPfannschmidt for the review and the heads up! Much appreciated 🙏🏻

File: pyproject.toml [hunk 2]
```diff
@@ -110,3 +110,4 @@ markers = ["network: mark tests that require internet access"]
 include = ["piptools*"]
 
 [tool.setuptools_scm]
+local_scheme = "dirty-tag"
```
Comment (webknjaz) [inline pyproject.toml:113]: I wouldn't make such changes because it looses the context in PR runs and during development.

**Pull request merged**